### PR TITLE
FEATURE: add full editing access to queued posts

### PR DESCRIPTION
**Pull request by xrav3nz**
> **Discussion & Screenshots:** https://meta.discourse.org/t/ability-to-change-the-category-of-a-post-before-approving/66754/7?u=xrav3nz

For pending new topics: the body of the post, title, categories and the tags are all editable.

For pending new replies: only the body is applicable and thus editable

Open to suggestions! Cheers 💪 

----

RE: "recording the edit" feature:

I have decided to split that into a separate PR because:

- keeping this PR small is easier for reviews
- it's really a different feature and out of the scope for this PR

### CODE REVIEW

File: app/assets/javascripts/discourse/templates/components/queued-post.hbs [hunk 1]
```diff
@@ -20,24 +20,41 @@
...
     </div>
 
+    {{#if showTags}}
+      <div class="list-tags">
+        {{#each tags as |t|}}
+          {{discourse-tag t}}
+        {{/each}}
+      </div>
+    {{else if editTags}}
+      {{tag-chooser tags=buffered.tags categoryId=buffered.category_id width='100%'}}
```
Comment (SamSaffron) [inline app/assets/javascripts/discourse/templates/components/queued-post.hbs]: hmmm should this not be handled by CSS? width 100% makes me nervous

**Comment by SamSaffron** [comment]
overall this PR is super tidy, can't wait to merge it in! thanks

File: app/assets/javascripts/discourse/templates/components/queued-post.hbs (hunk 1 above)
Comment (xrav3nz) [inline app/assets/javascripts/discourse/templates/components/queued-post.hbs]: I thought this is the intended way to [set the width on the tag chooser `select2` container](https://github.com/discourse/discourse/blob/130ae2cd4b6d60accde8267f51b0442ccd3217a5/app/assets/javascripts/discourse/components/tag-chooser.js.es6#L61)?

Without setting the width manually, the tag chooser is about 10px wider than everything else. `select2` internally sets the width with an inline style, which I cannot override without an `!important`.  /facepalm

p.s. The `100%` is wrapped by the parent `div.cooked`, and this is [set in the CSS.](https://github.com/discourse/discourse/blob/1146772debf861c5c8b6e4e3cebdba31a1138922/app/assets/stylesheets/desktop/queued-posts.scss#L18)

Comment (SamSaffron) [inline app/assets/javascripts/discourse/templates/components/queued-post.hbs]: oh my ... lucky @jjaffeux  is working on nuking select2 completely so this kind of hackery will no longer be needed ... he will sort this out as well when he gets to replacing the tag-chooser

**Comment by SamSaffron** [comment]
thanks heaps for this!

**Pull request merged**